client/src/actions/tagsFilter.js

```diff
@@ -0,0 +1,115 @@
+import {
+    FILTER_TAGS_SEARCH_PENDING, FILTER_TAGS_SEARCH_ERROR, FILTER_TAGS_SEARCH_SUCCESS,
+    FILTER_TAGS_LOAD_ERROR, FILTER_TAGS_LOAD_SUCCESS, FILTER_TAGS_LOAD_PENDING,
+    FILTER_MATERIALS_SEARCH_PENDING, FILTER_MATERIALS_SEARCH_SUCCESS, FILTER_MATERIALS_LOAD_PENDING,
+    FILTER_MATERIALS_LOAD_SUCCESS, FILTER_MATERIALS_LOAD_ERROR, FILTER_MATERIALS_SEARCH_ERROR
+} from '../constants';
+import * as QueryString from 'query-string';
+import { get } from '../utils/api-util';
+import { addError } from './errors';
+
+function searchPending(isMaterial) {
+    return {
+        type: isMaterial ? FILTER_MATERIALS_SEARCH_PENDING : FILTER_TAGS_SEARCH_PENDING
+    };
+}
+
+function searchSuccess(isMaterial, searchResults) {
+    return {
+        type: isMaterial ? FILTER_MATERIALS_SEARCH_SUCCESS : FILTER_TAGS_SEARCH_SUCCESS,
+        payload: {searchResults}
+    };
+}
+
+function searchError(isMaterial, error) {
+    return {
+        type: isMaterial ? FILTER_MATERIALS_SEARCH_ERROR : FILTER_TAGS_SEARCH_ERROR,
+        error: error
+    };
+}
+
+export function searchTags(queryString) {
```

Comment:
Just curious, but can't you specify what functions you want to export at the end of the file? If so, we could just list the exports at the bottom instead of in the front of the functions. If not, ignore this.

Reply:
Yes, you can export any variable i.e. `export searchTags;` or `export searchTags as search;` etc. But I prefer the inline syntax when possible